### PR TITLE
DecrementInteger mutator should not decrement zero indexes for array access

### PR DESCRIPTION
**Pull request by LoopKarma**
This PR:
- [x] Covered by tests
Fixes https://github.com/infection/infection/issues/1270

### CODE REVIEW

File: src/Mutator/Number/DecrementInteger.php [hunk 1]
```diff
@@ -98,6 +98,10 @@ private function isAllowedComparison(Node\Scalar\LNumber $node): bool
 
         $parentNode = ParentConnector::getParent($node);
 
+        if ($parentNode instanceof Node\Expr\ArrayDimFetch && $node->value === 0) {
```
Comment (maks-rafalko) [inline src/Mutator/Number/DecrementInteger.php:101]: I don't think this should be inside `isAllowedComparison()` method, because this is not related to `comparison`

Could you please move it e.g. before line 86 (`if ($this->isPartOfSizeComparison($node)) {`)

**Comment by maks-rafalko** [comment]
I pushed a fix for e2e tests, now CI is happy.

Thank you for this contribution, @LoopKarma!

**Comment by LoopKarma** [comment]
@maks-rafalko happy to help! Strange but I got errors in another e2e tests than Travis. How you came up that specifically this test should be improved?

**Comment by maks-rafalko** [comment]
> How you came up that specifically this test should be improved?

I ran `tests/e2e_tests bin/infection` and analyzed the output. There was something about difference between expected and actual output

**Comment by LoopKarma** [comment]
@maks-rafalko interesting, I ran the same command but got nothing for local php but got errors in other tests under docker setup. For sure I missed something here, since I'm pretty green here )
